.ci/pipeline/ci.yml

```diff
@@ -106,64 +77,96 @@ jobs:
     inputs:
       artifactName: 'documentation'
       targetPath: '$(Build.ArtifactStagingDirectory)/'
-- job: Linux_DPCPP
+- job: Linux
+  strategy:
+    matrix:
+      Python3.10_Sklearn1.0:
+        PYTHON_VERSION: '3.10'
+        SKLEARN_VERSION: '1.0'
+      Python3.10_Sklearn1.1:
+        PYTHON_VERSION: '3.10'
+        SKLEARN_VERSION: '1.1'
+      Python3.10_Sklearn1.2:
+        PYTHON_VERSION: '3.10'
+        SKLEARN_VERSION: '1.2'
+      Python3.11_Sklearn1.1:
+        PYTHON_VERSION: '3.11'
+        SKLEARN_VERSION: '1.1'
+      Python3.11_Sklearn1.2:
+        PYTHON_VERSION: '3.11'
+        SKLEARN_VERSION: '1.2'
   pool:
     vmImage: 'ubuntu-22.04'
   steps:
-  - task: UsePythonVersion@0
-    displayName: 'Use Python $(python.version)'
-  - script: |
-      wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
-      sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
-      rm GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
-      echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
-      sudo add-apt-repository -y "deb https://apt.repos.intel.com/oneapi all main"
-      sudo apt-get update
-      sudo apt-get install                                      \
-          intel-oneapi-common-vars                              \
-          intel-oneapi-common-licensing                         \
-          intel-oneapi-dpcpp-cpp-compiler                       \
-          intel-oneapi-dev-utilities                            \
-          intel-oneapi-libdpstd-devel
-      sudo bash -c 'echo libintelocl.so > /etc/OpenCL/vendors/intel-cpu.icd'
-      sudo mv -f /opt/intel/oneapi/compiler/latest/linux/lib/oclfpga /opt/intel/oneapi/compiler/latest/linux/lib/oclfpga_
+  - script: sudo apt-get update && sudo apt-get install -y libtinfo5 clang-format
     displayName: 'apt-get'
   - script: |
+      # TODO: update dpcpp version on next dal release
+      wget "https://github.com/intel/llvm/releases/download/sycl-nightly%2F20220614/dpcpp-compiler.tar.gz"
```

Comment:
This is opensource dpc compiler that differs from what is released by intel. I would say it's worth to stick with Intel releases to avoid unexpected problems